core/server/api/canary/members.js

```diff
@@ -192,7 +192,7 @@ module.exports = {
                 frame.options.withRelated = ['stripeSubscriptions'];
                 const member = await membersService.api.members.update(frame.data.members[0], frame.options);
 
-                const hasCompedSubscription = !!member.related('stripeSubscriptions').find(subscription => subscription.get('plan_nickname') === 'Complimentary');
+                const hasCompedSubscription = !!member.related('stripeSubscriptions').find(sub => sub.get('plan_nickname') === 'Complimentary' && sub.get('status') === 'active');
```

Comment:
@allouis Should the `active` check here also include other status values - `'active', 'trialing', 'unpaid', 'past_due'` -that we consider as an active subscription for paid member ? 🤔